test/parallel/test-buffer-write-noassert.js

```diff
@@ -11,12 +11,21 @@ function write(funx, args, result, res) {
     assert.deepStrictEqual(buf, res);
   }
 
+  /*{
+    const invalidArgs = Array.from(args);
+    invalidArgs[1] = -1;
+    assert.throws(
+      () => Buffer.alloc(9)[funx](...invalidArgs),
+      /^RangeError: Index out of range$/
+    );
+  }*/
+
   {
     const invalidArgs = Array.from(args);
     invalidArgs[1] = -1;
     assert.throws(
       () => Buffer.alloc(9)[funx](...invalidArgs),
-      /^RangeError: (?:Index )?out of range(?: index)?$/
+      /^RangeError(?:\[ERR_INDEX_OUT_OF_RANGE\])?: Index out of range$/
```

Comment:
I *think* this is the only regexp that should stay because we don't know ahead of time whether the error is coming from JS (if writing an int, for example) or C++ (if writing a float, for example).